compiler/optimizer/LocalOpts.cpp

```diff
@@ -8058,7 +8058,9 @@ TR_ColdBlockMarker::hasNotYetRun(TR::Node *node)
       TR::SymbolReference *symRef = node->getSymbolReference();
       bool isUnresolved;
 
-      if (comp()->compileRelocatableCode() && !comp()->getOption(TR_DisablePeekAOTResolutions))
+      if (comp()->compileRelocatableCode() &&
+          !comp()->getOption(TR_UseSymbolValidationManager) &&
```

Comment:
Are you just trying to preserve the old behaviour here by overloading on `TR_UseSymbolValidationManager`? The downstream code doesn't seem dependent on the symbol manager to me?

Reply:
> Are you just trying to preserve the old behaviour here by overloading on TR_UseSymbolValidationManager?

Yeah that's exactly it. I wanted there to be a way for the old code to run exactly as it did without all the new changes.

Reply:
Although... the `isInterpretedForHeuristics` change also affects the "old" behaviour... Though, it only affects it from a heuristic point of view and, if anything, results in better decisions being made, so I suppose (like https://github.com/eclipse/omr/pull/2978/commits/1e645c3d91ccae4927e6c7f1de50163d07145847 is just a generally good thing to have)